internal/core/commands/sync_dd_power_train_type.go

```diff
@@ -0,0 +1,201 @@
+	"github.com/rs/zerolog"
+	"github.com/volatiletech/sqlboiler/v4/queries/qm"
+	"gopkg.in/yaml.v3"
+)
+
+type SyncPowerTrainTypeCommand struct {
+}
+
+type SyncPowerTrainTypeCommandResult struct {
+	Status bool
+}
+
+func (*SyncPowerTrainTypeCommand) Key() string { return "SyncPowerTrainTypeCommand" }
+
+type SyncPowerTrainTypeCommandHandler struct {
+	DBS    func() *db.ReaderWriter
+	logger zerolog.Logger
+}
+
+func NewSyncPowerTrainTypeCommandHandler(dbs func() *db.ReaderWriter, logger zerolog.Logger) SyncPowerTrainTypeCommandHandler {
+	return SyncPowerTrainTypeCommandHandler{DBS: dbs, logger: logger}
+}
+
+func (ch SyncPowerTrainTypeCommandHandler) Handle(ctx context.Context, _ mediator.Message) (interface{}, error) {
+
+	all, err := models.DeviceDefinitions(models.DeviceDefinitionWhere.Verified.EQ(true),
+		qm.Load(models.DeviceDefinitionRels.DeviceStyles),
+		qm.Load(models.DeviceDefinitionRels.DeviceType),
+		qm.Load(models.DeviceDefinitionRels.DeviceMake)).All(ctx, ch.DBS().Reader)
+	if err != nil {
+		return nil, err
+	}
+
+	ch.logger.Info().Msgf("found %d device definitions verified, starting process...", len(all))
+	if len(all) == 0 {
+		return SyncPowerTrainTypeCommandResult{false}, nil
+	}
+
+	content, err := os.ReadFile("powertrain_type_rule.yaml")
+	if err != nil {
+		return nil, err
+	}
+
+	var powerTrainTypeData coremodels.PowerTrainTypeRuleData
+	err = yaml.Unmarshal(content, &powerTrainTypeData)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, definition := range all {
+		ch.logger.Info().Msgf("%s - Make:%s Model: %s Year: %d", definition.ID, definition.R.DeviceMake.NameSlug, definition.ModelSlug, definition.Year)
+
+		if definition.R.DeviceType == nil {
+			ch.logger.Error().Msgf("ID: %s with DeviceType is empty", definition.ID)
+			continue
+		}
+
+		metadataKey := definition.R.DeviceType.Metadatakey
+		var metadataAttributes map[string]any
+
+		if err := definition.Metadata.Unmarshal(&metadataAttributes); err == nil {
+			metadataAttributes = make(map[string]interface{})
+			metaData := make(map[string]interface{})
+
+			dt, err := models.DeviceTypes(models.DeviceTypeWhere.ID.EQ(definition.DeviceTypeID.String)).
+				One(ctx, ch.DBS().Reader)
+
+			if err != nil {
+				if !errors.Is(err, sql.ErrNoRows) {
+					return nil, err
+				}
+			}
+
+			var deviceTypeAttributes map[string][]coremodels.GetDeviceTypeAttributeQueryResult
+			if err := dt.Properties.Unmarshal(&deviceTypeAttributes); err == nil {
+				for _, deviceAttribute := range deviceTypeAttributes["properties"] {
+					metaData[deviceAttribute.Name] = deviceAttribute.DefaultValue
+				}
+			}
+
+			metadataAttributes[metadataKey] = metaData
+		}
+
+		for key, _ := range metadataAttributes[metadataKey].(map[string]any) {
+			if key == "powertrain_type" {
+				powerTrainType := ch.resolvePowerTrainTypeByMake(ctx, powerTrainTypeData, definition)
+				metadataAttributes[metadataKey].(map[string]interface{})["powertrain_type"] = powerTrainType
```

Comment:
Let's add a log message any time that the powertrain_type was previously set and we're changing it. Eg. "Updating powertrain_type from ICE to HEV."

Also, if the powertrain_type is unchanged, should we just skip to save the DB upsert call?